packages/react-native/local-cli/runMacOS/runMacOS.js

```diff
@@ -162,7 +162,7 @@ function buildProject(sourceDir, xcodeProject, scheme, args) {
   return new Promise((resolve, reject) => {
     const xcodebuildArgs = [
       xcodeProject.isWorkspace ? '-workspace' : '-project',
-      path.join(sourceDir, xcodeProject.path, xcodeProject.name),
+      path.join(sourceDir, xcodeProject.path || '.', xcodeProject.name),
```

Comment:
This needs to be backported to 0.74

Reply:
- **main:** https://github.com/microsoft/react-native-macos/pull/2237
- **0.74:** https://github.com/microsoft/react-native-macos/pull/2238
- **0.75:** https://github.com/microsoft/react-native-macos/pull/2239